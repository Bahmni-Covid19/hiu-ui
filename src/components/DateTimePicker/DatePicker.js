import React, { useState } from 'react';
import { KeyboardDatePicker } from '@material-ui/pickers';

const DatePicker = ({
  disableFuture,
  minDate,
  selectedDate,
  handleDateChange,
}) => (
  <KeyboardDatePicker
    autoOk="true"
    variant="inline"
    margin="normal"
    disableFuture={disableFuture}
    id="date-picker-dialog"
    minDate={minDate}
    format="dd/MM/yyyy"
    value={selectedDate}
    onChange={handleDateChange}
    KeyboardButtonProps={{
      'aria-label': 'change date',
    }}
  />
);

DatePicker.defaultProps = {
  minDate: {},
  disableFuture: false,
};

export default DatePicker;
