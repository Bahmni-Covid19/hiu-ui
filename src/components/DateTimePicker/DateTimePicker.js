import React, { useState } from "react";
import { KeyboardDateTimePicker } from "@material-ui/pickers";

const DateTimePicker = ({
  disableFuture,
  minDate,
  maxDate,
  selectedDate,
  handleDateChange
}) => {
  return (
    <KeyboardDateTimePicker
      margin="normal"
      disableFuture={disableFuture}
      id="date-picker-dialog"
      format="dd/MM/yyyy HH:mm"
      minDate={minDate}
      maxDate={maxDate}
      value={selectedDate}
      onChange={handleDateChange}
      KeyboardButtonProps={{
        "aria-label": "change date"
      }}
    />
  );
};

DateTimePicker.defaultProps = {
  disableFuture: false,
  minDate: {}
};

export default DateTimePicker;
