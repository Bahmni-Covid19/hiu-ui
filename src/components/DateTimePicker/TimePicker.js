import React, { useState } from "react";
import { KeyboardTimePicker } from "@material-ui/pickers";

const TimePicker = () => {
  const [selectedTime, setSelectedTime] = React.useState();

  const handleTimeChange = time => {
    setSelectedTime(time);
  };

  return (
    <KeyboardTimePicker
      margin="normal"
      id="time-picker"
      // label="Choose time"
      value={selectedTime}
      onChange={handleTimeChange}
      KeyboardButtonProps={{
        "aria-label": "change time"
      }}
    />
  );
};

export default TimePicker;
