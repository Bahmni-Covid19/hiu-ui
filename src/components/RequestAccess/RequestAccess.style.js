import styled from "styled-components";

const RequestAccessStyles = styled.div`
  h2 {
    color: #000000;
    font-size: 24px;
    line-height: 24px;
    font-weight: 300;
    padding: 20px;
    margin-bottom: 30px;
    border-bottom: 1px solid #000000;
  }
  .text-field-container {
    display: flex;
    align-items: center;
    margin-left: 20px;
    margin-bottom: 20px;
    .text-field-label {
      flex-basis: 20%;
      margin-right: 20px;
    }
  }
  .auto-complete {
    color: #009688;
    .text-field {
      color: #009688;
    }
  }
`;

export default RequestAccessStyles;
