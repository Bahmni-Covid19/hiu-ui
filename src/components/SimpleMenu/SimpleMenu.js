import React, { useEffect } from "react";
import Select from "@material-ui/core/Select";
import MenuItem from "@material-ui/core/MenuItem";
import SimpleMenuStyles from "./SimpleMenu.style";

const SimpleMenu = ({ menuItems, selectedValue, handleChange }) => {
  return (
    <SimpleMenuStyles>
      <Select
        id="select-menu"
        className="select-menu"
        value={selectedValue}
        onChange={handleChange}
      >
        {menuItems.map(type => (
          <MenuItem
            id="menu-item"
            key={type.value}
            value={type.value}
            onClick={handleChange}
          >
            {type.label}
          </MenuItem>
        ))}
      </Select>
    </SimpleMenuStyles>
  );
};

export default SimpleMenu;
