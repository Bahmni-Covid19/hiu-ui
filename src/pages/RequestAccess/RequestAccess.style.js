import styled from "styled-components";

const RequestAccessStyles = styled.div`
  h2 {
    color: #000000;
    font-size: 24px;
    line-height: 24px;
    font-weight: 300;
    padding: 20px 0;
    margin-bottom: 25px;
    border-bottom: 1px solid #000000;
    font-family: "Roboto", sans-serif;
  }
  .text-field-label {
    color: #000;
  }
  .label {
    display: block;
    color: #000;
    font-size: 16px;
    font-weight: 300;
    font-family: "Roboto", sans-serif;
  }
  .MuiGrid-container {
    padding: 10px 0;
  }
  .search-bar {
    vertical-align: center;
  }
  .auto-complete {
    color: #009688;
    .text-field {
      color: #009688;
    }
  }
  .create-consent-button {
    margin-top: 25px;
  }
  .error {
    display: block;
    padding: 20px 0 0px;
    color: #f44336;
    font-size: 16px;
    font-weight: 300;
    font-family: "Roboto", sans-serif;
  }
  .loader-container {
    position: fixed;
    z-index: 99;
    left: 0;
    top: 75px;
    background: #fff;
    width: 100%;
    height: 100%;
    align-items: center;
    display: flex;
    justify-content: center;
  }
`;

export default RequestAccessStyles;
