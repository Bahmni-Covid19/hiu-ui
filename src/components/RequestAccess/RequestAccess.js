import React, { useState, useEffect } from "react";
import { Button, CircularProgress } from "@material-ui/core";
import RequestAccessStyles from "./RequestAccess.style";
import SearchPatient from "../SearchPatient";
import DateTimePicker from "./../DateTimePicker/DateTimePicker";
import DatePicker from "./../DateTimePicker/DatePicker";
import Grid from "@material-ui/core/Grid";
import InputLabel from "@material-ui/core/InputLabel";
import RequestType from "./../RequestType/RequestType";
import SimpleMenu from "./../SimpleMenu/SimpleMenu";
import getNextDay from "./../../utils/getNextDay";
import purposeTypes from "./../../constants/puposeTypes";
import requestTypes from "./../../constants/requestTypes";

const RequestAccess = ({
  onCreateConsent,
  patientId,
  loading,
  success,
  error
}) => {
  const [selectedPurposeValue, setSelectedPurposeValue] = React.useState(
    purposeTypes[0].value
  );
  const [selectedStartDate, setSelectedStartDate] = React.useState(new Date());
  const [selectedEndDate, setSelectedEndDate] = React.useState(new Date());
  const [selectedRequestTypes, setselectedRequestTypes] = React.useState({
    PatientHistory: false,
    Medications: false,
    DiagnosisLab: false,
    RadiologyLab: false,
    Condition: false
  });

  const [selectedExpiryDate, setSelectedExpiryDate] = React.useState(
    getNextDay()
  );

  const handlePITypeChange = name => event => {
    setselectedRequestTypes({
      ...selectedRequestTypes,
      [name]: event.target.checked
    });
  };
  const handleStarteDateChange = date => {
    var startDate = date;
    setSelectedStartDate(startDate);
  };
  const handleEndDateChange = date => {
    var startDate = selectedStartDate;
    var endDate = date;
    if (startDate.getTime() < endDate.getTime()) {
      setSelectedEndDate(endDate);
    }
  };
  const handleExpiryDateChange = date => {
    setSelectedExpiryDate(date);
  };

  const handlePurposeSelection = event => {
    setSelectedPurposeValue(event.target.value);
  };

  return (
    <RequestAccessStyles>
      <h2>Consent request form</h2>
      <form>
        <span className="label">All the fields are mandatory.</span>
        {error && (
          <span className="error">Error occured while creating consent.</span>
        )}
        {success && (
          <span className="success">Consent created successfylly.</span>
        )}
        <Grid container spacing={3} alignItems="center">
          <Grid item xs={2}>
            <InputLabel className="text-field-label">
              Patient Identifier
            </InputLabel>
          </Grid>
          <Grid item xs={5}>
            <SearchPatient />
          </Grid>
        </Grid>
        <Grid container spacing={3} alignItems="center">
          <Grid item xs={2}>
            <InputLabel className="text-field-label">
              Purpose of request
            </InputLabel>
          </Grid>
          <Grid item xs={2}>
            <SimpleMenu
              menuItems={purposeTypes}
              handleChange={handlePurposeSelection}
              selectedValue={selectedPurposeValue}
            />
          </Grid>
        </Grid>
        <Grid container spacing={3} alignItems="center">
          <Grid item xs={2}>
            <InputLabel className="text-field-label"> Request from </InputLabel>
          </Grid>
          <Grid item xs={2}>
            <DatePicker
              disableFuture={true}
              handleDateChange={handleStarteDateChange}
              selectedDate={selectedStartDate}
            />
          </Grid>
        </Grid>
        <Grid container spacing={3} alignItems="center">
          <Grid item xs={2}>
            <InputLabel className="text-field-label"> Request to </InputLabel>
          </Grid>
          <Grid item xs={2}>
            <DatePicker
              disableFuture={true}
              minDate={selectedStartDate}
              handleDateChange={handleEndDateChange}
              selectedDate={selectedEndDate}
            />
          </Grid>
        </Grid>
        <Grid container spacing={3} alignItems="center">
          <Grid item xs={2}>
            <InputLabel className="text-field-label"> Request type </InputLabel>
          </Grid>
          <Grid item xs={10}>
            <RequestType
              requestTypes={requestTypes}
              handleChange={handlePITypeChange}
            />
          </Grid>
        </Grid>
        <Grid container spacing={3} alignItems="center">
          <Grid item xs={2}>
            <InputLabel className="text-field-label">Consent Expiry</InputLabel>
          </Grid>
          <Grid item xs={3}>
            <DateTimePicker
              handleDateChange={handleExpiryDateChange}
              selectedDate={selectedExpiryDate}
              disablePast="true"
            />
          </Grid>
        </Grid>
        <Grid container spacing={3} alignItems="center">
          <Button
            className="create-consent-button"
            variant="contained"
            color="primary"
            onClick={() => {
              onCreateConsent({
                patientId,
                selectedPurposeValue,
                selectedStartDate,
                selectedEndDate,
                selectedExpiryDate,
                selectedRequestTypes
              });
            }}
          >
            Request Consent
          </Button>
        </Grid>
      </form>
      {loading && (
        <div className="loader-container">
          <CircularProgress
            id="loader"
            className="loader"
            variant="indeterminate"
            disableShrink
            thickness={4}
          />
        </div>
      )}
    </RequestAccessStyles>
  );
};

SearchPatient.defaultProps = {
  success: false,
  error: false
};

export default RequestAccess;
