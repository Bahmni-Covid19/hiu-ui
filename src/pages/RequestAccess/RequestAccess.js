import React, { useState } from 'react';
import { Button, CircularProgress } from '@material-ui/core';
import Grid from '@material-ui/core/Grid';
import InputLabel from '@material-ui/core/InputLabel';
import RequestAccessStyles from './RequestAccess.style';
import SearchPatient from '../../components/SearchPatient';
import DateTimePicker from '../../components/DateTimePicker/DateTimePicker';
import DatePicker from '../../components/DateTimePicker/DatePicker';
import RequestType from '../../components/RequestType/RequestType';
import SimpleMenu from '../../components/SimpleMenu/SimpleMenu';
import getNextDay from '../../utils/getNextDay';

const RequestAccess = ({
  onCreateConsent, patientId, loading, error, purposesOfUse, hiTypes,
}) => {
  const [selectedStartDate, setSelectedStartDate] = useState(new Date());
  const [selectedEndDate, setSelectedEndDate] = useState(new Date());
  const usagePurposes = purposesOfUse.map((p) => ({ label: p.display, value: p.code }));
  const [selectedPurposeValue, setSelectedPurposeValue] = useState(
    usagePurposes.length > 0 ? usagePurposes[0].value : '',
  );
  const requestHiTypes = hiTypes.map((p) => ({ label: p.display, value: p.code }));
  const hiTypesInitialStates = Object.assign(...requestHiTypes.map((hiType) => ({ [hiType.value]: false })));
  const [selectedRequestTypes, setSelectedRequestTypes] = useState(hiTypesInitialStates);

  const [emptyPatientIDError, setEmptyPatientIDError] = useState(false);
  const [selectedExpiryDate, setSelectedExpiryDate] = useState(getNextDay());

  const isButtonEnabled = Object.values(selectedRequestTypes).some((x) => x);

  const handleHITypeChange = (name) => (event) => {
    setSelectedRequestTypes({
      ...selectedRequestTypes,
      [name]: event.target.checked,
    });
  };
  const handleStarteDateChange = (date) => {
    const startDate = date;
    setSelectedStartDate(startDate);
  };
  const handleEndDateChange = (date) => {
    const startDate = selectedStartDate;
    const endDate = date;
    if (startDate.getTime() < endDate.getTime()) {
      setSelectedEndDate(endDate);
    }
  };
  const handleExpiryDateChange = (date) => {
    setSelectedExpiryDate(date);
  };

  const handlePurposeSelection = (event) => {
    setSelectedPurposeValue(event.target.value);
  };

  return (
    <RequestAccessStyles>
      <h2>Consent request form</h2>
      <form>
        <span className="label">All the fields are mandatory.</span>
        {error && (
          <span className="error">Error occured while creating consent.</span>
        )}
        {emptyPatientIDError && (
          <span className="error">Please enter a patient identifier</span>
        )}
        <Grid container spacing={3} alignItems="center">
          <Grid item xs={2}>
            <InputLabel className="text-field-label">
              Patient Identifier
            </InputLabel>
          </Grid>
          <Grid item xs={5}>
            <SearchPatient />
          </Grid>
        </Grid>
        <Grid container spacing={3} alignItems="center">
          <Grid item xs={2}>
            <InputLabel className="text-field-label">
              Purpose of request
            </InputLabel>
          </Grid>
          <Grid item xs={2}>
            <SimpleMenu
              menuItems={usagePurposes}
              handleChange={handlePurposeSelection}
              selectedValue={selectedPurposeValue}
            />
          </Grid>
        </Grid>
        <Grid container spacing={3} alignItems="center">
          <Grid item xs={2}>
            <InputLabel className="text-field-label">
              Health info from
            </InputLabel>
          </Grid>
          <Grid item xs={2}>
            <DatePicker
              minDate="01/01/1900"
              disableFuture
              handleDateChange={handleStarteDateChange}
              selectedDate={selectedStartDate}
            />
          </Grid>
        </Grid>
        <Grid container spacing={3} alignItems="center">
          <Grid item xs={2}>
            <InputLabel className="text-field-label">Health info to</InputLabel>
          </Grid>
          <Grid item xs={2}>
            <DatePicker
              disableFuture
              minDate={selectedStartDate}
              handleDateChange={handleEndDateChange}
              selectedDate={selectedEndDate}
            />
          </Grid>
        </Grid>
        <Grid container spacing={3} alignItems="center">
          <Grid item xs={2}>
            <InputLabel className="text-field-label"> Request type </InputLabel>
          </Grid>
          <Grid item xs={10}>
            <RequestType
              requestTypes={requestHiTypes}
              handleChange={handleHITypeChange}
            />
          </Grid>
        </Grid>
        <Grid container spacing={3} alignItems="center">
          <Grid item xs={2}>
            <InputLabel className="text-field-label">Consent Expiry</InputLabel>
          </Grid>
          <Grid item xs={3}>
            <DateTimePicker
              handleDateChange={handleExpiryDateChange}
              selectedDate={selectedExpiryDate}
              disablePast="true"
            />
          </Grid>
        </Grid>
        <Grid container spacing={3} alignItems="center">
          <Button
            disabled={!isButtonEnabled}
            className="create-consent-button"
            variant="contained"
            color="primary"
            onClick={() => {
              if (!patientId) {
                setEmptyPatientIDError(true);
                return;
              }
              onCreateConsent({
                patientId,
                selectedPurposeValue,
                selectedStartDate,
                selectedEndDate,
                selectedExpiryDate,
                selectedRequestTypes,
              });
            }}
          >
            Request Consent
          </Button>
        </Grid>
      </form>
      {loading && (
        <div className="loader-container">
          <CircularProgress
            id="loader"
            className="loader"
            variant="indeterminate"
            disableShrink
            thickness={4}
          />
        </div>
      )}
    </RequestAccessStyles>
  );
};

SearchPatient.defaultProps = {
  success: false,
  error: false,
};

export default RequestAccess;
