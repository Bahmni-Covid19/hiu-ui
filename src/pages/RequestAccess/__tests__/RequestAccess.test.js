import React from "react";
import { shallow } from "enzyme";
import RequestAccess from "../RequestAccess";
import DatePicker from "../../../../src/components/DateTimePicker/DatePicker";
import DateTimePicker from "../../../../src/components/DateTimePicker/DateTimePicker";
import SimpleMenu from "../../../../src/components/SimpleMenu/SimpleMenu";
// import RequestType from "../../../../src/components/RequestType/RequestType";
// import requestTypes from "./../../../constants/requestTypes";

describe("Request Access", () => {
  let realDate;

  beforeEach(() => {
    const currentDate = new Date("2019-05-14T11:01:58.135Z");
    realDate = Date;
    global.Date = class extends Date {
      constructor(date) {
        if (date) {
          return super(date);
        }
        return currentDate;
      }
    };
  });

  afterEach(() => {
    // Cleanup
    global.Date = realDate;
  });

  it("Should render properly", () => {
    const wrapper = shallow(<RequestAccess />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(DatePicker).length).toEqual(2);
    expect(wrapper.find(DateTimePicker).length).toEqual(1);
  });

  it("Purpose of request, should have referral services selected by default", () => {
    const wrapper = shallow(<RequestAccess />);
    expect(wrapper.find(SimpleMenu).props().selectedValue).toEqual(
      "ReferralService"
    );
  });

  it("From: date component, should have current date when no date is selected ", () => {
    const wrapper = shallow(<RequestAccess />);
    expect(
      wrapper
        .find(DatePicker)
        .at(0)
        .props().selectedDate
    ).toEqual("2019-05-14T11:01:58.135Z");
  });

  it("To: date component, should have current date when no date is selected ", () => {
    const wrapper = shallow(<RequestAccess />);
    expect(
      wrapper
        .find(DatePicker)
        .at(1)
        .props().selectedDate
    ).toEqual("2019-05-14T11:01:58.135Z");
  });

  it("Expiry: date component, should have next day as current date when no date is selected ", () => {
    const wrapper = shallow(<RequestAccess />);
    const ActualExpiryDate = wrapper.find(DateTimePicker).props().selectedDate;
    const expectedExpiryDate = "2019-05-15T11:01:58.135Z";
    expect(ActualExpiryDate).toEqual(expectedExpiryDate);
  });
});
