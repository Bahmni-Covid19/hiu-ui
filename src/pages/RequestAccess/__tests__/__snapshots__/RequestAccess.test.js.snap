// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request Access Should render properly 1`] = `
<styled.div>
  <h2>
    Consent request form
  </h2>
  <form>
    <span
      className="label"
    >
      All the fields are mandatory.
    </span>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={2}
      >
        <WithStyles(ForwardRef(InputLabel))
          className="text-field-label"
        >
          Patient Identifier
        </WithStyles(ForwardRef(InputLabel))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={5}
      >
        <Connect(SearchPatient) />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={2}
      >
        <WithStyles(ForwardRef(InputLabel))
          className="text-field-label"
        >
          Purpose of request
        </WithStyles(ForwardRef(InputLabel))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={2}
      >
        <SimpleMenu
          handleChange={[Function]}
          menuItems={
            Array [
              Object {
                "label": "Referral Service",
                "value": "ReferralService",
              },
              Object {
                "label": "Break the Glass",
                "value": "BTG",
              },
            ]
          }
          selectedValue="ReferralService"
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={2}
      >
        <WithStyles(ForwardRef(InputLabel))
          className="text-field-label"
        >
          Health info from
        </WithStyles(ForwardRef(InputLabel))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={2}
      >
        <DatePicker
          disableFuture={true}
          handleDateChange={[Function]}
          minDate={1900-01-31T18:38:50.000Z}
          selectedDate={2019-05-14T11:01:58.135Z}
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={2}
      >
        <WithStyles(ForwardRef(InputLabel))
          className="text-field-label"
        >
          Health info to
        </WithStyles(ForwardRef(InputLabel))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={2}
      >
        <DatePicker
          disableFuture={true}
          handleDateChange={[Function]}
          minDate={2019-05-14T11:01:58.135Z}
          selectedDate={2019-05-14T11:01:58.135Z}
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={2}
      >
        <WithStyles(ForwardRef(InputLabel))
          className="text-field-label"
        >
           Request type 
        </WithStyles(ForwardRef(InputLabel))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={10}
      >
        <RequestType
          handleChange={[Function]}
          requestTypes={
            Array [
              Object {
                "label": "Condition",
                "value": "Condition",
              },
            ]
          }
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={2}
      >
        <WithStyles(ForwardRef(InputLabel))
          className="text-field-label"
        >
          Consent Expiry
        </WithStyles(ForwardRef(InputLabel))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={3}
      >
        <DateTimePicker
          disableFuture={false}
          disablePast={true}
          handleDateChange={[Function]}
          maxDate={2100-01-31T18:30:00.000Z}
          minDate={1900-01-31T18:38:50.000Z}
          selectedDate={2019-05-15T11:01:58.135Z}
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      spacing={3}
    >
      <WithStyles(ForwardRef(Button))
        className="create-consent-button"
        color="primary"
        disabled={true}
        onClick={[Function]}
        variant="contained"
      >
        Request Consent
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Grid))>
  </form>
</styled.div>
`;
